src/proxy/http/metrics/mod.rs

```diff
@@ -79,8 +87,19 @@ where
         Self {
             last_update: clock::now(),
             total: Counter::default(),
+            by_status: IndexMap::default(),
+        }
+    }
+}
+
+impl<C> Default for StatusMetrics<C>
+where
+    C: Hash + Eq,
+{
+    fn default() -> Self {
```

Comment:
Nit: This `Default` impl seems like it should be able to be derived. If there's a reason it can't be, it would be good to have a comment, I think.

Reply:
because C does not have to impl default